src/odemis/gui/win/acquisition.py

```diff
@@ -905,19 +907,32 @@ def _get_areas(self) -> List[Tuple[float]]:
         return: the bounding boxes (xmin, ymin, xmax, ymax in physical coordinates)
         of all areas to acquire.
         """
-
-        areas = get_tiled_areas(
-            pos=self.stage.position.value,
-            streams=self.get_acq_streams(),
-            tiles_nx=self.tiles_nx.value,
-            tiles_ny=self.tiles_ny.value,
-            overlap=self.overlap,
-            tiling_rng=self._tiling_rng,
-            selected_grids=self._selected_grids,
-            sample_centers=self._main_data_model.sample_centers,
-            rel_bbox=self._main_data_model.sample_rel_bbox,
-            whole_grid=self.whole_grid_chkbox.Value,
-        )
+        if not self.whole_grid_chkbox.Value:
+            area = get_stream_based_bbox(
+                        pos=self.stage.position.value,
+                        streams=self.get_acq_streams(),
+                        tiles_nx=self.tiles_nx.value,
+                        tiles_ny=self.tiles_ny.value,
+                        overlap=self.overlap,
+                        tiling_rng=self._tiling_rng,
+            )
+            # Cast to list, to have a consistent format with the alternative path
+            areas = [area] if area is not None else []
+        else:
+            if not hasattr(self._main_data_model, "sample_rel_bbox"):
```

Comment:
This should check sample_centers, because currently every cryo app has sample_rel_bbox: https://github.com/tmoerkerken/odemis/blob/f32a634a1faa233fbdafc00f7dab2a1712871436/src/odemis/gui/model/main_gui_data.py#L501. The whole_grid_chkbox gets hidden/disabled based on the sample_centers too
```suggestion
            if not self._main_data_model.sample_centers:
```